src/Functions/toRelativeYearNum.cpp

```diff
@@ -11,7 +11,41 @@ using FunctionToRelativeYearNum = FunctionDateOrDateTimeToSomething<DataTypeUInt
 
 REGISTER_FUNCTION(ToRelativeYearNum)
 {
-    factory.registerFunction<FunctionToRelativeYearNum>();
+    FunctionDocumentation::Description description_toRelativeYearNum = R"(
+Converts a date or date with time to the number of years elapsed since a certain fixed point in the past.
```

Comment:
The exact "certain fixed point in the past" is actually an implementation detail. It happens to be the year 0 but that could change in future. Maybe we should add a comment or note that users should not rely on that. See my comment to function `toDate` - basically: Function `toRelativeYear` must not be used stand-alone, it should only be used to compute relative year differences between two dates or dates with time.

The same applies to all `toRelative*` functions.